### DISCUSSION TITLE
Poisson model for gorilla sample plots

### DISCUSSION THREAD

**Discussion by dmi3kno** [inlabru-org/inlabru#206] [General]
I am trying to build a simple poisson model for counts for gorillas. I started by rasterizing counts to a somewhat coarse grid, but then I thought, perhaps I can reuse the sample plots provided in `gorillas_sf`. I am making a new column for counts>0 which I will need later on.

```{r}
library(tidyverse)
library(inlabru)
library(terra)
library(sf)

data(gorillas_sf)
nests <- gorillas_sf$nests
mesh <- gorillas_sf$mesh
boundary <- gorillas_sf$boundary
gcov <- gorillas_sf_gcov()
sample <- gorillas_sf$plotsample
sample_counts <- sample$counts |> 
  mutate(present=count>0)
```
The sample landplots are rather large
```{r}
ggplot() +
  geom_sf(data=boundary) +
  geom_sf(data=sample$plots) +
  geom_sf(data=nests, pch = "+", cex = 4, color="blue", alpha=0.1)+
  geom_sf(data=sample$nests, pch = "+", cex = 4, color = "red") +
  geom_sf_label(data=sample_counts,aes(label=count)) +
  labs(x = "Easting", y = "Northing")
```
So if I were to build a mesh off of the plot centroids it would probably look like this

```{r}
px_mesh <- fm_mesh_2d_inla(loc=st_as_sfc(sample_counts),
                           boundary = boundary,
                           max.edge = c(0.5, 1), 
                           crs=st_crs(sample_counts))

px_matern <- INLA::inla.spde2.pcmatern(px_mesh, 
  prior.sigma = c(0.1, 0.01),
  prior.range = c(0.1, 0.01)
)

ggplot()+
  geom_fm(data=px_mesh)+
  geom_sf(data=sample$counts, aes(color=count))+
  theme_minimal()
```
![000012](https://github.com/inlabru-org/inlabru/assets/13419011/5e765eda-95e5-4f0f-87be-bdb08b08eb59)

This is about right that on average there's space for only 1 location between the nodes, because the landplots from which the counts are collected appear rather big.

But of course, the model fails to converge

```{r}
comps <-  ~ field(geometry, model=px_matern) + Intercept(1) 

fit_poi <- bru(comps,
              like(family = "poisson", data=sample_counts,
                   formula = count~field+Intercept,
                   E=exposure,
                   domain = list(geometry = px_mesh)))
summary(fit_poi)

#>  *** inla.core.safe:  rerun to try to solve negative eigenvalue(s) in the Hessian 
```

Whats the right way of fitting a counts model if my survey plots are rather big and I want to predict the counts on the landplots in between the known location. Does my problem stem from the fact that the land is not divided into equal pixels for accumulation of counts?

The poisson model ran fine when I had all counts accumulated to the pixels of about 0.8 ha.

**Comment by finnlindgren** [comment]
The plotsample data is an artificially created example to use for illustrating principles. How to implement real models depend on the specific type of plot sampling was involved, and the relative spatial scales of the plot sizes, domain of interest, and spatial correlation range of the latent variables and covariates. [Edited] The https://inlabru-org.github.io/inlabru/articles/2d_lgcp_plotsampling.html article illustrates the case when regions have been exhaustively searched, so that locations are known, and not only counts.

When the survey regions are large compared with the needed mesh resolution, using the plot midpoints only is a bad approximation of the correctly aggregated model that has $E(count_i) = \int_{plot_i} \lambda(s) ds$.
The aggregation example towards the end of https://www.maths.ed.ac.uk/~flindgre/talks/Lindgren_RSS2023.pdf is close to what you need; `bru_mapper_logsumexp()` and a `fm_int(mesh, plots)` call.  @enoch26  is doing his PhD on models involving these constructions.

[Edited] I'm not sure why your model crashes though; the mesh does have some issues along the inner boundary; try it without specifying the plot centroids in the mesh construction; they're not needed there, as they don't need to match mesh vertices.
